spec/unit/http/client_spec.rb

```diff
@@ -85,7 +85,10 @@ def expect_http_error(cause, expected_message)
 
   context "for GET requests" do
     it "includes default HTTP headers" do
-      stub_request(:get, uri).with(headers: {'X-Puppet-Version' => /./, 'User-Agent' => /./})
+      stub_request(:get, uri).with do |request|
+        expect(request.headers).to include({'X-Puppet-Version' => /./, 'User-Agent' => /./})
+        expect(request.headers).to_not include('X-Puppet-Profiling')
+      end
```

Comment:
rspec has a nice way of ensuring a collection contains exactly the specified items and no more:

```ruby
        expect(request.headers).to contain_exactly({'X-Puppet-Version' => /./, 'User-Agent' => /./})
```

Reply:
:) Yeah, this felt like the best way to care about these specific headers rather than include all the others in the test